File: logica-matematica/ejercicio-023/resoluciones/lucas-pajarito/torneo.js
const equipos = [
    {
        nombre: "Dragons",
        puntaje: 95
    },
    {
        nombre: "Titans",
        puntaje: 82
    },
    {
        nombre: "Phoenix",
        puntaje: 88
    },
    {
        nombre: "Warriors",
        puntaje: 91
    },
    {
        nombre: "Legends",
        puntaje: 84
    }
];

function calcularEstadisticas(equipos) {

    console.log("=== TORNEO DE ESPORTS ===");

    let suma = 0;
    const puntajes = [];

    equipos.forEach(equipo => {

        suma += equipo.puntaje;
        puntajes.push(equipo.puntaje);

        console.log(`Equipo: ${equipo.nombre}`);
        console.log(`Puntaje: ${equipo.puntaje}`);
        console.log("-------------------------------");

    });

    const promedio = suma / equipos.length;

    puntajes.sort((a, b) => a - b);

    let mediana;

    if (puntajes.length % 2 === 0) {

        const mitad = puntajes.length / 2;

        mediana = (puntajes[mitad - 1] + puntajes[mitad]) / 2;

    } else {

        mediana = puntajes[Math.floor(puntajes.length / 2)];

    }

    console.log(`Promedio: ${promedio.toFixed(2)}`);
    console.log(`Mediana: ${mediana}`);

}

calcularEstadisticas(equipos);

File: logica-matematica/ejercicio-003/resoluciones/lucas-pajarito/torneo.py
equipos = [
    {
        "nombre": "Dragons",
        "partidos_jugados": 12,
        "victorias": 9,
        "derrotas": 3
    },
    {
        "nombre": "Titans",
        "partidos_jugados": 12,
        "victorias": 7,
        "derrotas": 5
    },
    {
        "nombre": "Phoenix",
        "partidos_jugados": 12,
        "victorias": 10,
        "derrotas": 2
    },
    {
        "nombre": "Warriors",
        "partidos_jugados": 12,
        "victorias": 5,
        "derrotas": 7
    },
    {
        "nombre": "Legends",
        "partidos_jugados": 12,
        "victorias": 11,
        "derrotas": 1
    }
]


def calcular_porcentaje_victorias(equipos):

    print("=== RENDIMIENTO DE LOS EQUIPOS ===")

    for equipo in equipos:

        porcentaje = (equipo["victorias"] / equipo["partidos_jugados"]) * 100

        print(f'Equipo: {equipo["nombre"]}')
        print(f'Partidos jugados: {equipo["partidos_jugados"]}')
        print(f'Victorias: {equipo["victorias"]}')
        print(f'Porcentaje de victorias: {porcentaje:.2f}%')
        print("-------------------------------")


calcular_porcentaje_victorias(equipos)